azurelinuxagent/common/protocol/goal_state.py

```diff
@@ -352,35 +410,37 @@ def _parse_plugin_settings(ext_handler, plugin_settings):
         runtime_settings_node = find(settings[0], "RuntimeSettings")
         seqNo = getattrib(runtime_settings_node, "seqNo")
         runtime_settings_str = gettext(runtime_settings_node)
-        try:
-            runtime_settings = json.loads(runtime_settings_str)
-        except ValueError as e:
-            logger.error("Invalid extension settings")
-            return
+        if runtime_settings_str is not None:
+            try:
+                runtime_settings = json.loads(runtime_settings_str)
+            except ValueError as e:
+                logger.error("Invalid extension settings")
+                return
 
         depends_on_level = 0
         depends_on_node = find(settings[0], "DependsOn")
-        if depends_on_node != None:
+        if depends_on_node is not None:
             try:
                 depends_on_level = int(getattrib(depends_on_node, "dependencyLevel"))
             except (ValueError, TypeError):
                 logger.warn("Could not parse dependencyLevel for handler {0}. Setting it to 0".format(name))
                 depends_on_level = 0
 
-        for plugin_settings_list in runtime_settings["runtimeSettings"]:
-            handler_settings = plugin_settings_list["handlerSettings"]
-            ext = Extension()
-            # There is no "extension name" in wire protocol.
-            # Put
-            ext.name = ext_handler.name
-            ext.sequenceNumber = seqNo
-            ext.publicSettings = handler_settings.get("publicSettings")
-            ext.protectedSettings = handler_settings.get("protectedSettings")
-            ext.dependencyLevel = depends_on_level
-            thumbprint = handler_settings.get(
-                "protectedSettingsCertThumbprint")
-            ext.certificateThumbprint = thumbprint
-            ext_handler.properties.extensions.append(ext)
+        if runtime_settings is not None:
```

Comment:
the previous code assumed this property was always present and would have raised an exception if it wasn't. What are the implications of adding this check?

Reply:
If I recall, I added this based on the Windows tests (since that agent supports multiconfig). In that case, we'll receive an ExtensionRuntimeSettings instead. Granted, the Linux agent doesn't support that yet, so I can remove this check + the accompanying unit test if you prefer.